did_doc_sov/src/lib.rs

```diff
@@ -164,7 +164,12 @@ impl Serialize for DidDocumentSov {
 
         for service_sov in &self.services {
             let service: Service<ExtraFieldsSov> = service_sov.clone().try_into().map_err(serde::ser::Error::custom)?;
-            builder = builder.add_service(service);
+            // Not very efficient, but
```

Comment:
perhaps one more way to consider:
Change `DidDocumentSov` such instead of 
```
services: Vec<ServiceSov>
```
it would use HashMap, using service id as key. That way:
- you have to worry about duplicates at serialization
- you could give user option to update a service, though this might come off as untransparent API, as builder's `add_service` would be possibly not appending but overwriting (if service with same `id` exists) (so then `add_service` should probably be renamed)
- guarantee that serialized diddoc does not have 2 services with same `id`

Some possible drawbacks:
- if you read a diddoc which has service with 2 duplicate services, incorrectly with the same ID (perhaps we should fail at deserialization in a first place), then you would end up "squashing them together" and produced serialized diddoc would not be the same as the invalid original.

I think the current fix is sufficient, just something to consider - perhaps for separate PR if you think such a change make sense.

Reply:
You have to realize why `DidDocumentSov` stores `Vec<ServiceSov>`. It has a very good reason, which is that the infallible getter `services`, must return `&[ServiceSov]` (as I was explaining to you).

Reply:
Another issue is that from `HashMap` you get values only in arbitrary order (which is the reason I got rid of it in the builder), with O(capacity) complexity. Then the deserialization -> serialization -> deserialization test would be failing due to arbitrariness in the ordering.